### PR TITLE
add test for service controller sync and remove unused function

### PR DESCRIPTION
**Pull request by NickrenREN**

### CODE REVIEW

**Comment by NickrenREN** [comment]
@k8s-bot ok to test

File: pkg/controller/service/servicecontroller_test.go [hunk 1]
```diff
@@ -280,3 +280,52 @@ func TestGetNodeConditionPredicate(t *testing.T) {
...
+	if keyExpected != keyGot.(string) {
+		t.Fatalf("get service key error, expected: %s, got: %s", keyExpected, keyGot.(string))
+	}
+
+	err := controller.syncService(keyExpected)
+	if err != nil {
+		t.Fatalf("sync service error: %v", err)
+	}
+
+	_, exsit := controller.cache.get(keyExpected)
```
Comment (derekwaynecarr) [inline pkg/controller/service/servicecontroller_test.go]: nit: typo, should be exist

**Review by derekwaynecarr — CHANGES_REQUESTED** [review]
one minor nit to update.

**Comment by NickrenREN** [comment]
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[pkg/controller/service/servicecontroller_test.go, line 326 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/39944#-KaayXzllzJgq9Xth0YC-r1-326:-KbweadA-V_z0_6kg2U1:bxoqmz4) ([raw file](https://github.com/kubernetes/kubernetes/blob/73fb34c431af80df33732627f436b70a58a64379/pkg/controller/service/servicecontroller_test.go#L326)):*
<details><summary><i>Previously, derekwaynecarr (Derek Carr) wrote…</i></summary><blockquote>

nit: typo, should be exist
</blockquote></details>

Done.  Thanks

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/39944)*

**Comment by NickrenREN** [comment]
@k8s-bot unit test this

**Comment by NickrenREN** [comment]
@k8s-bot unit test this

File: pkg/controller/service/servicecontroller_test.go (hunk 1 above)
Comment (NickrenREN) [inline pkg/controller/service/servicecontroller_test.go]: Oh, Thanks @derekwaynecarr . Modified

**Comment by NickrenREN** [comment]
@derekwaynecarr  Sorry to bother you. Do you have time to review the test codes again? Thanks

**Comment by NickrenREN** [comment]
@derekwaynecarr  @mikedanese  PTAL. Thanks a lot

**Review by derekwaynecarr — APPROVED** [review]

**Comment by derekwaynecarr** [comment]
looks like bazel needs an update.

can you update and poke?

**Comment by NickrenREN** [comment]
@k8s-bot unit test this

**Comment by NickrenREN** [comment]
@derekwaynecarr  I have updated the codes and tested locally. Have no idea why unit test failed

root@nickren-ubuntu14:/home/nickren/GoProjects/src/k8s.io/kubernetes# make test WHAT=./pkg/controller/service/
Running tests for APIVersion: v1,apps/v1beta1,authentication.k8s.io/v1,authentication.k8s.io/v1beta1,authorization.k8s.io/v1,authorization.k8s.io/v1beta1,autoscaling/v1,autoscaling/v2alpha1,batch/v1,batch/v2alpha1,certificates.k8s.io/v1beta1,extensions/v1beta1,imagepolicy.k8s.io/v1alpha1,policy/v1beta1,rbac.authorization.k8s.io/v1beta1,rbac.authorization.k8s.io/v1alpha1,settings.k8s.io/v1alpha1,storage.k8s.io/v1beta1,storage.k8s.io/v1,federation/v1beta1
+++ [0313 10:39:10] Running tests without code coverage
ok      k8s.io/kubernetes/pkg/controller/service        0.064s

root@nickren-ubuntu14:/home/nickren/GoProjects/src/k8s.io/kubernetes# make test WHAT=./pkg/controller/service KUBE_GOFLAGS="-v" KUBE_TEST_ARGS='-run ^TestSyncService$'
Running tests for APIVersion: v1,apps/v1beta1,authentication.k8s.io/v1,authentication.k8s.io/v1beta1,authorization.k8s.io/v1,authorization.k8s.io/v1beta1,autoscaling/v1,autoscaling/v2alpha1,batch/v1,batch/v2alpha1,certificates.k8s.io/v1beta1,extensions/v1beta1,imagepolicy.k8s.io/v1alpha1,policy/v1beta1,rbac.authorization.k8s.io/v1beta1,rbac.authorization.k8s.io/v1alpha1,settings.k8s.io/v1alpha1,storage.k8s.io/v1beta1,storage.k8s.io/v1,federation/v1beta1
+++ [0313 11:13:19] Running tests without code coverage
=== RUN   TestSyncService
--- PASS: TestSyncService (0.00s)
PASS
ok      k8s.io/kubernetes/pkg/controller/service        0.066s

**Comment by NickrenREN** [comment]
@k8s-bot unit test this

**Comment by NickrenREN** [comment]
@derekwaynecarr Done. PTAL ,thanks in advance.

**Comment by NickrenREN** [comment]
@derekwaynecarr  Done. PTAL. Thanks in advance

**Comment by NickrenREN** [comment]
@k8s-bot unit test this

**Comment by NickrenREN** [comment]
@k8s-bot unit test this

**Comment by NickrenREN** [comment]
@mikedanese  Thanks for your lgtm label. But Jenkins unit/integration failed. Rebase the codes. PTAL

**Comment by NickrenREN** [comment]
@k8s-bot unit test this

**Comment by NickrenREN** [comment]
@k8s-bot unit test this

**Comment by NickrenREN** [comment]
@k8s-bot bazel test this

**Comment by NickrenREN** [comment]
@k8s-bot bazel test this

**Comment by NickrenREN** [comment]
ping @mikedanese  for approval

**Comment by mikedanese** [comment]
@k8s-bot ok to test

**Comment by NickrenREN** [comment]
@k8s-bot verify test this

**Comment by NickrenREN** [comment]
Thanks @mikedanese

**Pull request merged**